internal/oauth/oauth.go

```diff
@@ -58,13 +58,36 @@ const (
 	GovClientID = "0oabtyfelbTBdoucy297" // GovClientID for production
 )
 
-func changeHostNameToContainer() {
-	if !strings.Contains(config.UserAgent, config.ContainerHostName) {
-		config.UserAgent = strings.ReplaceAll(config.UserAgent, config.HostName, config.ContainerHostName)
-		config.HostName = config.ContainerHostName
+// patchConfigHostnameFromEnvs patches the agent hostname based on set env vars.
+func patchConfigHostnameFromEnvs(checker envs.EnvChecker) {
+	var builder strings.Builder
+	if checker.IsPopulated(config.AtlasActionHostNameEnv) {
+		appendToHostName(&builder, config.AtlasActionHostName)
+	}
+	if checker.IsPopulated(config.GitHubActionsHostNameEnv) {
+		appendToHostName(&builder, config.GitHubActionsHostName)
+	}
+	if checker.IsPopulated(config.ContainerizedHostNameEnv) {
+		appendToHostName(&builder, config.DockerContainerHostName)
+	}
```

Comment:
[optional] I wonder if we should always append, either the value itself, or a `-`, if value not populated.
I'm thinking about the scenario when only one of the 3 values is populated, looking at the hostname, can we know which env var it represents?
wdyt?

Reply:
So the hostName if the CLI is run natively would be "-|-|-"?
It's included in the userAgent so at that point I wonder whether we would just consider extending the UserAgent for each possible hostname

ie: 
`UserAgent      = fmt.Sprintf("%s/%s (%s;%s;%s)", ToolName, version.Version, runtime.GOOS, runtime.GOARCH, HostName)`

changes to
`UserAgent      = fmt.Sprintf("%s/%s (%s;%s;%s;%s;%s)", ToolName, version.Version, runtime.GOOS, runtime.GOARCH, MaybeAtlasActionHostName, MaybeGitHubActionsHostName, MaybeDockerContainerHostName)`

I think it starts to get a bit messy either way so I'd slightly favour keeping it as is, but don't feel particularly strongly if you'd like a change

Reply:
indeed, when all 3 values are not populated, the "-|-|-" is not helpful, so in this case the hostname should not be set to this value. 
I was thinking more about the scenario one at least one is, like "-|-|sameValue". Might be easier to parse / interpret this value rather than "someValue" (which might be AtlasActionHostName OR GitHubActionsHostName OR  DockerContainerHostName).
This is a [nit], so I'm ok with keeping it as it is, or extending the userAgent, as long as this won't have side effects or create confusion when interpreting the telemetry data

Reply:
Sure I'll make this change

Reply:
Updated the tests to describe the different scenarios
Let me know if that's good with you